### PR TITLE
Add F5-TTS with semantic token training results

### PR DESCRIPTION
**Pull request by yuekaizhang**
The PR replaced the text tokens input in F5-TTS with pretrained cosyvoice2 semantic tokens. 
Or you can think of it as replacing the unet based flow matching model in the [cosyvoice2](https://arxiv.org/abs/2412.10117) with DiT.

Summary of the training results using wenetspeech4tts data：

|        Model                               | Seed-TTS test_zh CER  | Training Steps |
|---------------------------------------|--------------------|-----------------|
| f5-tts           | 3.02% (16 steps)  | 300k  |
| **f5-tts-semantic-token** |  **1.79% (16 steps)**   | 50k |

### CODE REVIEW

**Comment by JinZr** [comment]
huge reduction on CER, i'll look into this pr.

File: egs/wenetspeech4tts/TTS/prepare.sh [hunk 1]
```diff
@@ -139,3 +139,28 @@ if [ $stage -le 6 ] && [ $stop_stage -ge 6 ]; then
       touch data/fbank/.${prefix}_split.done
   fi
 fi
+
+if [ $stage -le 7 ] && [ $stop_stage -ge 7 ]; then
+  log "Stage 7: Extract cosyvoice2 FSQ token (used by ./f5-tts semantic token experiment)"
+  pip install s3tokenizer
```
Comment (JinZr) [inline egs/wenetspeech4tts/TTS/prepare.sh:145]: i would prefer adding an additional `requirements.txt` down the recipe directory as the manner adopted by the `valle` and `matcha` recipe.

**Comment by yfyeung** [comment]
> The PR replaced the text tokens input in F5-TTS with pretrained cosyvoice2 semantic tokens. Or you can think of it as replacing the unet based flow matching model in the [cosyvoice2](https://arxiv.org/abs/2412.10117) with DiT.
> 
> Summary of the training results using wenetspeech4tts data：
> 
> Model	Seed-TTS test_zh CER	Training Steps
> f5-tts	3.02% (16 steps)	300k
> **f5-tts-semantic-token**	**1.79% (16 steps)**	50k

An interesting and impressive result! I am currently working on replacing MEL with CosyVoice2 tokens, i.e., discrete-E2TTS.

**Comment by Benioh** [comment]
Is the model before F5 the same as CosyVoice2? If so, I believe the main capability comes from the language model , so it would be better to compare the results with CosyVoice2 to assess the improvement brought by Dit FlowMatching. Comparing it with F5 doesn’t make much sense.

File: egs/wenetspeech4tts/TTS/prepare.sh (hunk 1 above)
Comment (yuekaizhang) [reply egs/wenetspeech4tts/TTS/prepare.sh]: Done. Thanks.

**Comment by yuekaizhang** [comment]
> Is the model before F5 the same as CosyVoice2? If so, I believe the main capability comes from the language model , so it would be better to compare the results with CosyVoice2 to assess the improvement brought by Dit FlowMatching. Comparing it with F5 doesn’t make much sense.

Yes, the PR's results used pretrained Cosyvoice2 LLM to predict semantic tokens during inference time. Updated the README.md also to avoid the confusion.

**Review by JinZr — APPROVED** [review]
LGTM, thank you for the PR!
